### PR TITLE
f-content-cards@v3.0.0-beta.1  - Update content cards to become a renderless component

### PR DESCRIPTION
**Pull request by matthewhardern**
In order to both reducing the ongoing complexity around implementing and the maintainability of content cards component, after a POC it was decided that we would now move to an approach where content cards purely served as a scoped slot component that provided all the functions around content cards and delivering the cards array, but be agnostic in terms of style. This approach will now enable us to implement different styles of the content cards being displayed without bloating this component. 

The changes included in this PR represent discussion had at the end of 2020 on the approach, removing the previous editions concept of layouts in the component, leaving this now entirely up to the consumer. For an insight into the discussion please see #547 now closed pull request.

I have updated the unit tests accordingly and removed no longer needed tests ( example grouping tests are now gone, as this would be handled by the consumer ).

I will be submitting a future PR with some major changes to the story book stories, I have not included this as part of this PR due to its size. Also attached to this next PR will be an update to the readme.

- [x] Unit tests have been [created|updated]

### CODE REVIEW

File: packages/components/organisms/f-content-cards/src/components/cardTemplates/GroupHeaderCard.vue [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        {{ title }}
+    </h2>
+</template>
+
+<script>
+export default {
+    name: 'GroupHeaderCard',
+    props: {
+        title: {
+            default: '',
```
Comment (kevinrodrigues) [inline packages/components/organisms/f-content-cards/src/components/cardTemplates/GroupHeaderCard.vue]: Do you need a default if it's required?

File: packages/components/organisms/f-content-cards/src/components/cardTemplates/GroupHeaderCard.vue [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            required: true,
+            type: String
+        }
+    }
+};
+</script>
+
+<style lang="scss" module>
+    .c-contentCards--group-title {
```
Comment (kevinrodrigues) [inline packages/components/organisms/f-content-cards/src/components/cardTemplates/GroupHeaderCard.vue]: Minor but could you tab this in and line break between selectors please.

File: packages/components/organisms/f-content-cards/src/components/cardTemplates/GroupHeaderCard.vue (hunk 1 above)
Comment (matthewhardern) [reply packages/components/organisms/f-content-cards/src/components/cardTemplates/GroupHeaderCard.vue]: agreed, will change

File: packages/components/organisms/f-content-cards/src/components/ContentCards.js [hunk 3]
```diff
@@ -224,36 +114,13 @@ export default {
         cards (current, previous) {
             this.$emit('get-card-count', current.length);
 
-            if (current.length && (current.length !== previous.length) && !this.groupCards) {
+            if (current.length && (current.length !== previous.length)) {
```
Comment (ashleynolan) [inline packages/components/organisms/f-content-cards/src/components/ContentCards.js:117]: Not specific to this line, but quickly scanning this, it's not super obvious what `current` and `previous` represent. Would be worth updating the JSDoc comment to explain what it holds (I'm guessing it a current set of content cards being displayed vs a previous set, but I wouldn't be 100% sure) and I wouldn't know what the context of current compared with previous would be (previous page load?)

**Review by ashleynolan — APPROVED** [review]
Looks good – one small comment but other than that LGTM

File: packages/components/organisms/f-content-cards/src/components/ContentCards.js (hunk 3 above)
Comment (matthewhardern) [reply packages/components/organisms/f-content-cards/src/components/ContentCards.js:117]: Happy to update the comment, are you ok with that update being in the docs PR following this?

**Review by kevinrodrigues — APPROVED** [review]

File: packages/components/organisms/f-content-cards/src/components/ContentCards.js (hunk 3 above)
Comment (ashleynolan) [reply packages/components/organisms/f-content-cards/src/components/ContentCards.js:117]: Yeh, absolutely 👍

**Pull request merged**